From 29daec4d9af19b819d5eedb33af904519657b6ef Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 11 Aug 2026 21:04:31 +0100
Subject: [PATCH 1/3] modalias: Don't set icon to 'emblem-system-symbolic' on
 matched hardware

It's unsure whether this was originally added as a hint when the h/w
matched or it's a bug where it was set as a fallback icon
unconditionally even when an appstream icon exists.

Either way the behaviour can be confusing so remove it.
---
 plugins/modalias/gs-plugin-modalias.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/plugins/modalias/gs-plugin-modalias.c b/plugins/modalias/gs-plugin-modalias.c
index 3d334c210..81d7ed1ec 100644
--- a/plugins/modalias/gs-plugin-modalias.c
+++ b/plugins/modalias/gs-plugin-modalias.c
@@ -147,9 +147,6 @@ refine_app (GsPluginModalias            *self,
 		items = as_provided_get_items (prov);
 		for (guint j = 0; j < items->len; j++) {
 			if (gs_plugin_modalias_matches (self, (const gchar*) g_ptr_array_index (items, j))) {
-				g_autoptr(GIcon) ic = NULL;
-				ic = g_themed_icon_new ("emblem-system-symbolic");
-				gs_app_add_icon (app, ic);
 				gs_app_add_quirk (app, GS_APP_QUIRK_NOT_LAUNCHABLE);
 				break;
 			}
-- 
GitLab


From c20680a69bac2769d0f358adc03f46c7bdad054a Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 11 Aug 2026 21:09:16 +0100
Subject: [PATCH 2/3] Add quirk GS_APP_QUIRK_HARDWARE_MATCHED when modaliases
 match the system

When kind AS_COMPONENT_KIND_DRIVER provided modaliases match the system
add a quirk to mark it as such.
---
 lib/gs-app.h                          | 2 ++
 plugins/modalias/gs-plugin-modalias.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/lib/gs-app.h b/lib/gs-app.h
index d59eee4e4..1938ee4c9 100644
--- a/lib/gs-app.h
+++ b/lib/gs-app.h
@@ -138,6 +138,7 @@ typedef enum {
  * @GS_APP_QUIRK_HIDE_EVERYWHERE:	The app should not be shown anywhere (it’s blocklisted)
  * @GS_APP_QUIRK_DO_NOT_AUTO_UPDATE:	The app should not be automatically updated
  * @GS_APP_QUIRK_FROM_DEVELOPMENT_REPOSITORY: The app is from a development/beta repository (Since: 49)
+ * @GS_APP_QUIRK_HARDWARE_MATCHED:	The app provides support for hardware attached to this system (Since: 52)
  *
  * The application attributes.
  **/
@@ -162,6 +163,7 @@ typedef enum {
 	GS_APP_QUIRK_HIDE_EVERYWHERE	= 1 << 16,	/* Since: 3.36 */
 	GS_APP_QUIRK_DO_NOT_AUTO_UPDATE	= 1 << 17,	/* Since: 3.36 */
 	GS_APP_QUIRK_FROM_DEVELOPMENT_REPOSITORY = 1 << 18,	/* Since: 49 */
+	GS_APP_QUIRK_HARDWARE_MATCHED   = 1 << 19,	/* Since: 52 */
 	GS_APP_QUIRK_LAST  /*< skip >*/
 } GsAppQuirk;
 
diff --git a/plugins/modalias/gs-plugin-modalias.c b/plugins/modalias/gs-plugin-modalias.c
index 81d7ed1ec..ab948e7ea 100644
--- a/plugins/modalias/gs-plugin-modalias.c
+++ b/plugins/modalias/gs-plugin-modalias.c
@@ -147,6 +147,7 @@ refine_app (GsPluginModalias            *self,
 		items = as_provided_get_items (prov);
 		for (guint j = 0; j < items->len; j++) {
 			if (gs_plugin_modalias_matches (self, (const gchar*) g_ptr_array_index (items, j))) {
+				gs_app_add_quirk (app, GS_APP_QUIRK_HARDWARE_MATCHED);
 				gs_app_add_quirk (app, GS_APP_QUIRK_NOT_LAUNCHABLE);
 				break;
 			}
-- 
GitLab


From 4db5660266223299ac562e52717b1fcfa7d14f01 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 11 Aug 2026 21:11:21 +0100
Subject: [PATCH 3/3] Add hint & tooltip when h/w drivers match the current
 system

---
 src/gs-summary-tile.c  | 14 ++++++++++++++
 src/gs-summary-tile.ui | 25 ++++++++++++++++++++-----
 2 files changed, 34 insertions(+), 5 deletions(-)

diff --git a/src/gs-summary-tile.c b/src/gs-summary-tile.c
index 7c17f6f31..084c8ac9c 100644
--- a/src/gs-summary-tile.c
+++ b/src/gs-summary-tile.c
@@ -70,6 +70,7 @@ struct _GsSummaryTile
 	GtkWidget	*name;
 	GtkWidget	*summary;
 	GtkWidget	*bin;
+	GtkWidget	*matches_hardware_icon;
 	GtkWidget	*stack;
 	gint		 preferred_width;
 
@@ -91,6 +92,7 @@ gs_summary_tile_refresh (GsAppTile *self)
 	GsAppIconsState app_icons_state;
 	GsApp *app = gs_app_tile_get_app (self);
 	gboolean installed;
+	gboolean matches_hardware;
 	g_autofree gchar *name = NULL;
 	const gchar *summary;
 
@@ -164,6 +166,16 @@ gs_summary_tile_refresh (GsAppTile *self)
 
 	gtk_widget_set_visible (tile->bin, installed);
 
+	if (gs_app_has_quirk (app, GS_APP_QUIRK_HARDWARE_MATCHED)) {
+		matches_hardware = TRUE;
+	} else {
+		matches_hardware = FALSE;
+	}
+
+	gtk_widget_set_visible (tile->matches_hardware_icon, matches_hardware);
+	gtk_widget_set_tooltip_text (tile->matches_hardware_icon,
+				     matches_hardware ? _("Known to support your system") : NULL);
+
 	if (name != NULL) {
 		gtk_accessible_update_property (GTK_ACCESSIBLE (tile),
 						GTK_ACCESSIBLE_PROPERTY_LABEL, name,
@@ -284,6 +296,8 @@ gs_summary_tile_class_init (GsSummaryTileClass *klass)
 					      summary);
 	gtk_widget_class_bind_template_child (widget_class, GsSummaryTile,
 					      bin);
+	gtk_widget_class_bind_template_child (widget_class, GsSummaryTile,
+					      matches_hardware_icon);
 	gtk_widget_class_bind_template_child (widget_class, GsSummaryTile,
 					      stack);
 }
diff --git a/src/gs-summary-tile.ui b/src/gs-summary-tile.ui
index 55b400d7c..b08b82f8f 100644
--- a/src/gs-summary-tile.ui
+++ b/src/gs-summary-tile.ui
@@ -96,11 +96,26 @@
                           <object class="GtkStackPage">
                             <property name="name">image</property>
                             <property name="child">
-                              <object class="GtkImage" id="image">
-                                <property name="pixel-size">64</property>
-                                <style>
-                                  <class name="icon-dropshadow"/>
-                                </style>
+                              <object class="GtkOverlay">
+                                <child>
+                                  <object class="GtkImage" id="image">
+                                    <property name="pixel-size">64</property>
+                                    <style>
+                                      <class name="icon-dropshadow"/>
+                                    </style>
+                                  </object>
+                                </child>
+                                <child type="overlay">
+                                  <object class="GtkImage" id="matches_hardware_icon">
+                                    <property name="visible">False</property>
+                                    <property name="halign">start</property>
+                                    <property name="valign">end</property>
+                                    <property name="margin-end">2</property>
+                                    <property name="margin-bottom">2</property>
+                                    <property name="pixel-size">22</property>
+                                    <property name="icon-name">info-symbolic</property>
+                                  </object>
+                                </child>
                               </object>
                             </property>
                           </object>
-- 
GitLab

